### ISSUE TITLE
Integration with fzf

### ISSUE THREAD

**Issue by tapyu** [atanunq/viu#109]
Does anybody know how to integrate the viu with `fzf`? My goal it to make `fzf` print the pdf cover image in the `preview` section. At the moment, my humble code is

```shell
function fpp()
{
  # open PDF using fuzzy finder with preview
  local open=xdg-open   # this will open pdf file withthe default PDF viewer on KDE, xfce, LXDE and perhaps on other desktops.
  rg --type 'pdf' --files | fzf  --preview-window down:80% --preview '
      pdftoppm -f 1 -l 1 -singlefile -jpeg -tiffcompression jpeg -- {1} | viu -
      # viu {1}
      '
}
```

It prompts to
![image](https://user-images.githubusercontent.com/22801918/209212468-da9d1bab-95ba-432b-b19c-fa08a2250d47.png)

Any help is welcome :)

**Comment by tapyu** [comment]
When I use `viu -b` it works though, what is happening?

**Comment by IceAsteroid** [comment]
Yeah, integration with fzf while having the original resolutions of images would be perfect.

It works with -b option to force block output though but not with the original resolutions.

**Comment by sweetbbak** [comment]
its kind of both but major factor is fzf. The thing is, is that this works perfectly fine with Ueberzug (which is now out of development but is still the only working option) kitty +kitten icat used to work until last month on kitty terminal when they rebuilt icat to be faster. Fzf draws another buffer and cant display image directly. 

It sucks because there is almost nothing that is like fzf that also has the option for previews, let alone image previews. A lot of the choice menus dont give you the index of the item you are on so you cant really make it work in a hacky way. Its weird because I'm not sure how ueberzug is still the only one that works when it was developed as a stop-gap years ago. It seems like this is a very common need but theres nothing that really fills that gap right now, I've been looking for a long time.

**Comment by tapyu** [comment]
@sweetbbak Why/How did `Ueberzug` make it work? AFAIK `Ueberzug` is permanently out due to harassment... I used to use `ueberzug` too.

**Comment by sweetbbak** [comment]
yea, they are done for good with ueberzug, which is really unfortunate. It does still work though and there is a new project called Ueberzugpp in C++ that is reviving it. It intends to be a drop in replacement for ubz but I was having a few issues compared to the original project when it came to use with fzf. Mainly the image would freeze and not change but I also need more time to figure it out. 

I just started using the original one since icat broke with fzf. There are a couple of other projects that are using the Kitty protocol (with kitty and St) and Unicode character placeholders which may fix the issue as well. Right now just using the original ueberzug is the only way I've been able to get it to work. Fzf can only read the escape codes as strings and cant interpret them to images, ubz display over the top of fzf all together

**Comment by ghuh** [comment]
I'm trying to use `viu` to get image previews for `fzf` in Ghostty and it looks like `viu -b` is still the only working option.

Has anyone found a better solution?
